File: app/src/main/java/eu/napcode/gonoteit/repository/user/UserRepository.java
package eu.napcode.gonoteit.repository.user;

import com.apollographql.apollo.api.Response;

import eu.napcode.gonoteit.AuthenticateMutation;
import io.reactivex.Observable;

public interface UserRepository {

    Observable<Response<AuthenticateMutation.Data>> authenticateUser(String login, String password);

    public void saveUserAuthData(String login, String token);
}

File: app/src/main/java/eu/napcode/gonoteit/repository/user/UserRepository.kt
package eu.napcode.gonoteit.repository.user

import android.arch.lifecycle.LiveData
import android.arch.lifecycle.MutableLiveData
import com.apollographql.apollo.api.Response

import eu.napcode.gonoteit.AuthenticateMutation
import eu.napcode.gonoteit.dao.user.UserEntity
import eu.napcode.gonoteit.model.UserModel
import eu.napcode.gonoteit.repository.Resource
import io.reactivex.Completable
import io.reactivex.Observable

interface UserRepository {

    fun isUserLoggedIn(): Boolean

    fun updateUserFromRemote(authResource: MutableLiveData<Resource<AuthenticateMutation.Data>>?)

    fun getLoggedInUser(): LiveData<UserModel?>

    fun authenticateUser(login: String, password: String, authResponse : MutableLiveData<Resource<AuthenticateMutation.Data>>): Observable<Response<AuthenticateMutation.Data>>

    fun logoutUser(): Completable
}
